app/models/conversation_creator.rb

```diff
@@ -0,0 +1,45 @@
+class ConversationCreator
+  def self.create(author:, recipient_username:, subject:, message_params: nil)
+    new.create(
+      author: author,
+      recipient_username: recipient_username,
+      subject: subject,
+      message_params: message_params,
+    )
+  end
+
+  def create(author:, recipient_username:, subject:, message_params: nil)
+    Conversation.create(
+      author: author,
+      recipient: recipient(recipient_username),
+      subject: subject,
+    ).tap do |conversation|
+      ensure_recipient(conversation: conversation, username: recipient_username)
+      if conversation.persisted?
+        MessageCreator.create(
+          conversation: conversation,
+          author: author,
+          body: message_params[:body],
+          hat_id: message_params[:hat_id],
+          create_modnote: create_modnote?(message_params),
+        )
+      end
+    end
+  end
+
+private
+
+  def recipient(username)
+    @_recipient ||= User.find_by(username: username)
```

Comment:
Is there an underscore on this instance variable because it's not used directly?

Reply:
Correct, it's a convention I picked up.